lib/philomena_web/controllers/api/json/posts_controller.ex

```diff
@@ -0,0 +1,66 @@
+defmodule PhilomenaWeb.Api.Json.PostController do
+  use PhilomenaWeb, :controller
+
+  alias PhilomenaWeb.PostJson
+  alias Philomena.Posts.Post
+  alias Philomena.Repo
+  import Ecto.Query
+
+  def show(conn, %{"forum_id" => forum_id, "topic_id" => topic_id, "id" => post_id}) do
+    post = 
+      Post
+      |> where(id: ^post_id)
+      |> where(destroyed_content: false)
+      |> join(:inner, [p], _ in assoc(p, :topic))
+      |> where([_p, t], t.hidden_from_users == false)
+      |> where([_p, t], t.slug == ^topic_id)
+      |> join(:inner, [_p, t], _ in assoc(t, :forum))
+      |> where([_p, _t, f], f.access_level == "normal")
+      |> where([_p, _t, f], f.short_name == ^forum_id)      
+      |> preload([:user, :topic])
+      |> Repo.one()
+
+    cond do
+      is_nil(post) ->
+        conn
+        |> put_status(:not_found)
+        |> text("")
+
+      true ->
+        json(conn, %{post: PostJson.as_json(post)})
+    end
+  end
+
+  def index(conn, %{"forum_id" => forum_id, "topic_id" => topic_id} = params) do
+    page = paginate(params["page"])
+    posts = 
+      Post
+      |> where(destroyed_content: false)
+      |> join(:inner, [p], _ in assoc(p, :topic))
+      |> where([_p, t], t.hidden_from_users == false)
+      |> where([_p, t], t.slug == ^topic_id)
+      |> join(:inner, [_p, t], _ in assoc(t, :forum))
+      |> where([_p, _t, f], f.access_level == "normal")
+      |> where([_p, _t, f], f.short_name == ^forum_id)      
+      |> where([p], p.topic_position >= ^(25 * (page - 1)) and p.topic_position < ^(25 * page))
+      |> order_by(asc: :topic_position)
+      |> preload([:user, :topic])
+      |> Repo.all()
+
+    json(conn, %{posts: Enum.map(posts, &PostJson.as_json/1), page: page})
```

Comment:
Should there be a way to see the total number of pages?

Reply:
I'm unsure how to get that without doing an entire other query each time. 

Most sane API clients should see when there's less than 25 results and automagically stop poking for more.

Reply:
perhaps something like `%{total: topic.post_count}` (this would be total post count, not total pages, but would be consistent with the image search endpoint)

Reply:
Okay, got it done.